File: spk/API/Entities/UserLicenseAccess.cs
namespace API.Entities;

public class UserLicenseAccess : BaseEntity
{
    public string UserId { get; set; } = string.Empty;

    public AppUser? User { get; set; }

    public Guid LicenseId { get; set; }

    public License? License { get; set; }

    public DateTime StartDate { get; set; } = DateTime.UtcNow;

    public DateTime? EndDate { get; set; }

    public bool IsActive { get; set; } = true;

    public AccessSource AccessSource { get; set; } = AccessSource.Admin;

    public bool IsDemoAccess { get; set; }

    public bool GrantedAutomatically { get; set; }

    public DateTime? ExpiresAt { get; set; }
}

File: Client/src/models/userLicenseAccess.ts
export type AccessSource = 1 | 2 | 3 | 4 | 5

export interface UserSummary {
  id: string
  email: string
  displayName: string
  roles: string[]
}

export interface UserLicenseAccess {
  id: string
  userId: string
  userEmail: string
  licenseId: string
  licenseName: string
  startDate: string
  endDate?: string | null
  isActive: boolean
  isCurrentlyActive: boolean
  accessSource: AccessSource
  createdAt: string
}

export interface MyLicenseAccess {
  licenseId: string
  licenseName: string
  hasAccess: boolean
  startDate?: string | null
  endDate?: string | null
  accessSource?: AccessSource | null
  isDemoAccess?: boolean
  grantedAutomatically?: boolean
  expiresAt?: string | null
}

export interface CreateUserLicenseAccess {
  userId: string
  licenseId: string
  startDate: string
  endDate?: string | null
  isActive: boolean
  accessSource: AccessSource
}

export interface UpdateUserLicenseAccess {
  startDate: string
  endDate?: string | null
  isActive: boolean
  accessSource: AccessSource
}
